tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs

```diff
@@ -138,6 +138,31 @@ private static string GetEnvironmentVariableIfIsNotEmpty(string key, string defa
             return defaultValue;
         }
 
+        private static void SetEnvironmentVariablesIfNotEmpty(Dictionary<string, string> dictionary, params string[] keys)
+        {
+            if (dictionary is null || keys is null)
+            {
+                return;
+            }
+
+            foreach (var key in keys)
+            {
+                if (key is null)
+                {
+                    continue;
+                }
+
+                var value = EnvironmentHelpers.GetEnvironmentVariable(key);
+                if (!string.IsNullOrEmpty(value))
+                {
+                    lock (dictionary)
```

Comment:
In the original calling code we don't `lock` before setting the values. Do we really need th e lock here? There's no concurrency here is there? 🤔

Reply:
True, done in https://github.com/DataDog/dd-trace-dotnet/pull/3719/commits/947ddfacdcdc0d58214c27e6900f7c3552861a7c